TestingLowerBounds/Kernel/Monoidal.lean

```diff
@@ -117,8 +117,16 @@ lemma swap_copy : (swap α α) ∘ₖ (copy α) = copy α := by
     Measure.dirac_apply' _ hs]
   congr
 
+@[simp]
+lemma swap_swap : (swap α β) ∘ₖ (swap β α) = kernel.id := by
+  ext ab s hs
+  rw [comp_apply, Measure.bind_apply hs (kernel.measurable _), id_apply, swap_apply,
+    lintegral_dirac' _ (kernel.measurable_coe _ hs), swap_apply, Prod.swap_swap]
```

Comment:
This proof is fine, but there is another possible proof that does not use `ext`, and I would like to be sure that we have the lemmas to do it:
- the composition of two deterministic kernels is the deterministic kernel of the composition of the functions
- swap composed with swap is the identity
- the deterministic kernel for the identity function is kernel.id

Do we have the lemma for that first step?

Reply:
I think we don't have it, I cannot find it in the API docs and `exact?` does not seem to be able to prove it. I proved it and put it in `TestingLowerBounds.Kernel.Basic`, since in that file there is also `deterministic_prod_deterministic`.